### PR TITLE
Organizing our documentation and readme

### PR DESCRIPTION
**Pull request by tbombach**
Adding `PULL_REQUEST_TEMPLATE.md`, `CONTRIBUTING.md` and moving some info from `README.md` to those pages.

Adding a pattern that makes it easier for service partners to know where they are in the process and how to get back to our documentation repo.
- `PULL_REQUEST_TEMPLATE.md`: automatically gets added to a pull request when it's opened. Contains checkboxes that the author needs to check so help make sure the PR is ready to be reviewed. [Preview this file in my fork](https://github.com/tbombach/azure-rest-api-specs/blob/pr-template/.github/PULL_REQUEST_TEMPLATE.md)
- `CONTRIBUTING.md`: puts our contribution guidelines in a standard place that is consistent with where other Github repos put their guidelines. This shows up before a PR is added and gives a contributor expectations for the Clone->Edit->PR->Review->Merge process. [Preview this file in my fork](https://github.com/tbombach/azure-rest-api-specs/blob/pr-template/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by tbombach** [comment]
@kirthik I've organized our documentation a bit and added a lot of info. Please let me know if there's anything that should be changed before this gets merged (or any basic things you think should be added). If there are bigger things to add, I'll do those in another PR since there's value in getting the contribution guidelines and pull request template checked in right away.

Since I consolidated all the review criteria/linter rules to one documentation page here, I'll be able to remove it from AutoRest repo and just link to this one :tada:

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (amarzavery) [inline .github/PULL_REQUEST_TEMPLATE.md]: Quality of Swagger

Comment (tbombach) [inline .github/PULL_REQUEST_TEMPLATE.md]: Resolved. I was trying to do a checklist like "Quality Swagger..." "conforms to the specification", "follows the patterns described...", but that was needlessly confusing. I changed it so each list item is a full sentence.

**Review by kirthik — APPROVED** [review]
I did not mean to block this PR on my comments.  Please take a look at  my comments and address them if you can. Feel free to merge and address them later too.

File: .github/CONTRIBUTING.md
Comment (tbombach) [inline .github/CONTRIBUTING.md]: Really good point, that causes some confusion for some teams when I direct them to adx-documentation-pr and they ping me since they don't have access. I'll add that before merging

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tbombach) [inline .github/PULL_REQUEST_TEMPLATE.md]: Yep, will do

File: documentation/review-criteria.md
Comment (tbombach) [inline documentation/review-criteria.md]: Good point, I'll reorganize this. It think the swagger-good-patterns could be reorganized to make it easier to find the patterns, but that'll be my next check-in. Doesn't make sense for me to duplicate the content.

**Comment by tbombach** [comment]
Removed `review-criteria.md`, since it's either in documentation (`linter.md` and `swagger-good-patterns.md`) or in the `CONTRIBUTING.md` guidelines to explain the process.

**Pull request merged**
